sources/System/Controllable/ControllableSystem.cpp

```diff
@@ -18,8 +19,11 @@ namespace BBM
 	{
 		auto &controllable = entity.get<ControllableComponent>();
 		auto &movable = entity.get<MovableComponent>();
+		auto health = entity->tryGetComponent<HealthComponent>();
 		Vector2f move = controllable.move.normalized() * controllable.speed;
 
+		if (health->getHealthPoint() <= 0)
```

Comment:
Same as last comment, this is illegal. If you want to disable the component, you can remove it.